File: telegram-bot/utils/formatters.py
from datetime import datetime
from typing import Optional
from config import CURRENCY_SYMBOLS

def format_currency(amount: float, currency: str = "NGN") -> str:
    symbol = CURRENCY_SYMBOLS.get(currency, currency)
    return f"{symbol}{amount:,.2f}"

def format_date(date_str: Optional[str] = None) -> str:
    if date_str:
        try:
            dt = datetime.fromisoformat(date_str.replace('Z', '+00:00'))
            return dt.strftime("%Y-%m-%d %H:%M")
        except:
            return date_str
    return datetime.now().strftime("%Y-%m-%d %H:%M")

def calculate_progress_bar(current: float, target: float, length: int = 10) -> str:
    if target <= 0:
        return "▱" * length

    progress = min(current / target, 1.0)
    filled = int(progress * length)
    return "▰" * filled + "▱" * (length - filled)

def format_percentage(value: float) -> str:
    return f"{value:.1f}%"

File: frontend/src/utils/formatters.js
export const CURRENCY_SYMBOLS = {
  NGN: '₦',
  USD: '$',
  EUR: '€'
}

export const formatCurrency = (amount, currency = 'NGN') => {
  const symbol = CURRENCY_SYMBOLS[currency] || currency
  return `${symbol}${parseFloat(amount).toLocaleString('en-US', { minimumFractionDigits: 2, maximumFractionDigits: 2 })}`
}

export const formatDate = (dateString) => {
  if (!dateString) return ''
  const date = new Date(dateString)
  return date.toLocaleDateString('en-US', {
    year: 'numeric',
    month: 'short',
    day: 'numeric',
    hour: '2-digit',
    minute: '2-digit'
  })
}

export const calculateProgress = (current, target) => {
  if (target <= 0) return 0
  return Math.min((current / target) * 100, 100)
}
